tests/functional/RuleMatchedLog.php

```diff
@@ -0,0 +1,182 @@
+
+class RuleMatchedLog extends DbTestCase
+{
+    public function testCriteriaAddition()
+    {
+        $printer = new Printer();
+        $rulematchedlog = new GlobalRuleMatchedLog();
+        // Addition test
+        $xml = '<?xml version="1.0" encoding="UTF-8" ?>
+        <REQUEST>
+          <CONTENT>
+            <DEVICE>
+              <INFO>
+                <COMMENTS>Imprimante HP LaserJet Pro MFP M428fdw</COMMENTS>
+                <ID>123456</ID>
+                <MANUFACTURER>HP</MANUFACTURER>
+                <MEMORY>64</MEMORY>
+                <MODEL>LaserJet Pro MFP M428fdw</MODEL>
+                <NAME>Imprimante HP LaserJet Pro MFP M428fdw</NAME>
+                <RAM>128</RAM>
+                <SERIAL>ABC123456</SERIAL>
+                <TYPE>PRINTER</TYPE>
+                <UPTIME>7 days, 12:34:56.78</UPTIME>
+                <IPS><IP>192.168.1.100</IP></IPS>
+                <MAC>01:23:45:67:89:ab</MAC>
+              </INFO>
+              <PORTS>
+                <PORT>
+                  <IFDESCR>Ethernet/1</IFDESCR>
+                  <IFINERRORS>0</IFINERRORS>
+                  <IFINOCTETS>1234567890</IFINOCTETS>
+                  <IFINTERNALSTATUS>1</IFINTERNALSTATUS>
+                  <IFLASTCHANGE>12.34 seconds</IFLASTCHANGE>
+                  <IFMTU>1500</IFMTU>
+                  <IFNAME>Port 1</IFNAME>
+                  <IFNUMBER>1</IFNUMBER>
+                  <IFOUTERRORS>0</IFOUTERRORS>
+                  <IFOUTOCTETS>987654321</IFOUTOCTETS>
+                  <IFSPEED>1000000000</IFSPEED>
+                  <IFSTATUS>1</IFSTATUS>
+                  <IFTYPE>7</IFTYPE>
+                  <IP>192.168.1.100</IP>
+                  <IPS>
+                    <IP>192.168.1.100</IP>
+                  </IPS>
+                  <MAC>01:23:45:67:89:ab</MAC>
+                </PORT>
+              </PORTS>
+            </DEVICE>
+          </CONTENT>
+          <QUERY>SNMP</QUERY>
+          <DEVICEID>bar</DEVICEID>
+        </REQUEST>';
+
+        $converter = new \Glpi\Inventory\Converter();
+        $data = $converter->convert($xml);
+        $json = json_decode($data);
+
+        $inventory = new \Glpi\Inventory\Inventory($json);
+        $date_add = $_SESSION['glpi_currenttime'];
+
+        if ($inventory->inError()) {
+            $this->dump($inventory->getErrors());
+        }
+        $this->boolean($inventory->inError())->isFalse();
+        $this->array($inventory->getErrors())->isEmpty();
+
+        $printers_id = $inventory->getItem()->fields['id'];
+        $this->integer($printers_id)->isGreaterThan(0);
+
+        $this->boolean($printer->getFromDB($printers_id))->isTrue();
+
+        $this->boolean($rulematchedlog->getFromDBByCrit(
+            [
+                'items_id' => $printers_id,
+                'itemtype' => Printer::class,
+            ]
+        ))->isTrue();
+        $criteria = $rulematchedlog->fields['criteria'];
+        $criteria = str_replace('"_auto":1,', '', $criteria);
```

Comment:
Could you add a comment to explain why you need to replace the specific value ?
It might help future readers to know why this is done :)

Reply:
_auto should be kept, as is used by rules (like other criteria)